tests/unit_tests/util/fake_server.py

```diff
@@ -93,9 +93,9 @@ def __init__(self, loop: asyncio.AbstractEventLoop):
                         }
                     }
                 )
-        # If the session cookie is invalid the OAuth Proxy will serve the
-        # provider sign-in page (status = 200) instead of a 401 Unauthorized
-        return web.Response(status=200, text="OAuth provider sign-in page")
+            resp = {"data": {"message": "Bad request"}}
```

Comment:
Should this be within an if block?

```suggestion
            elif self._get_query_header(request) == "bad":
                resp = {"data": {"message": "Bad request"}}
                return web.json_response(resp, status=400)
            ...
```